pubtools/pulplib/_impl/model/distributor.py

```diff
@@ -26,6 +26,9 @@ class Distributor(PulpObject):
     of type `yum_distributor` may be used to create yum repositories.
     """
 
+    repo_id = pulp_attrib(type=str, default=None, pulp_field="repo_id")
```

Comment:
With addition of this new attribute, I think a new validator in Repository class is appropriate to ensure the field is correct.

What I mean is, right now it's possible for bogus data to be created like this:

```
repo = Repository(id="some-id", distributors=(Distributor(id="dist-id", repo_id="oops-some-other-id"))
```

It would be best if such code raised an exception rather than allowing the invalid object to be created.  The validation would be something like: on each distributor, if repo_id is not None then it must match the ID of this repository.

Could you please look into that?

Reply:
Sure. In the comments above, it's mentioned that distributor is always going to include repo_id. Should this exclude checking if repo_id is not None then compare. It should compare distributor.repo_id with repo.id in all cases.

Reply:
I think the validation should be: for each Distributor on a Repository, if the distributor repo_id is not None, then it must be equal to the Repository's id.  So: yes, None should be tolerated.